specutils/analysis/template_match.py

```diff
@@ -0,0 +1,105 @@
+from ..spectra.spectrum1d import Spectrum1D
+from ..spectra.spectrum_collection import SpectrumCollection
+from ..manipulation import FluxConservingResampler
+import numpy as np
+
+def _normalize(observed_spectrum, template_spectrum):
```

Comment:
(minor) maybe call this something slightly more specific like `_normalize_for_template_matching`?  Goes with my suggestion of generifying the file name a bit so that we share this with other related use cases.